src/board/devices/output/7seg.ts

```diff
@@ -0,0 +1,205 @@
+import { Device } from 'board/devices/device'
+import { Byte, Halfword, Word } from '../../../types/binary'
+enum SEVENsegNr {
+  _0011_1111 = 0,
+  _0000_0110 = 1,
+  _0101_1011 = 2,
+  _0100_1111 = 3,
+  _0110_0110 = 4,
+  _0110_1101 = 5,
+  _0111_1101 = 6,
+  _0000_0111 = 7,
+  _0111_1111 = 8,
+  _0110_1111 = 9,
+  _0111_0111 = 10, //a
+  _0111_1100 = 11, //b
+  _0100_1100 = 12, //c
+  _0101_1110 = 13, //d
+  _0111_1001 = 14, //e
+  _0111_0001 = 15 //f
+}
+
+export class SEVENseg extends Device {
+  public startAddress = Word.fromUnsignedInteger(0x60000110)
+  public endAddress = Word.fromUnsignedInteger(0x60000115)
+  public startAddressBin = Word.fromUnsignedInteger(0x60000114)
+  public endAddressBin = Word.fromUnsignedInteger(0x60000115)
+  public isReadOnly = false
+  public isVolatile = true
+
+  private static readonly MAX_SEG_NUMBER: number = 31
+  private oldBin: Halfword = this.memory.readHalfword(this.startAddressBin)
+  private oldSeg: Word = this.memory.readWord(this.startAddress)
+  private displays: boolean[][] = [
```

Comment:
This would mean that I have to store if the binary or "standard" memory section is the "newer" value. And I hoped to save calculation time this way

Reply:
But don't you have already this information? You have your `isBin` and `isReg` methods and with that you know from which source you can get your data if I understood correctly.

Reply:
What I do not understand is: You have this saved arrays but you never update them anywhere besides in the `getDisplay()` method where the array is updated and returned. The only place they are directly accessed without updating is inside the `isOn()` method but this method is never called anywhere. And the way it is implemented now the array is only updated if the `getDisplay()` method is called.

I would keep the array as it is but I would update the array whenever the write methods of the device are called. And if I'm not wrong then you would not even need the `isBin()` and `isReg()` methods inside the `getDisplay()` method because you would just simply return the array that is requested from the UI. The array is just updated by whatever was last called.

For example:
1. In the first cycle the user writes a `Byte` to memory at position `0x60000114` which would be the binary interface. This would mean array 0 and 1 are updated with the true and false flags.
2. In the next clock cycle the user writes a `Byte` to memory to address `0x60000110`, the raw interface which means the array 0 needs to be updated according to the ones and zeros in the byte.
3. During another cycle the user writes a Halfword at position `0x60000114` which means array 0-4 need to be updated.
4. And so on and so on...

So the array is always kept up to date.